diff --git a/dist/index.js b/dist/index.js
index 87e7a1a2841bab7a801fda5ad4ecd2536a5158b7..69a4a48d744ce1d3dbebbebb8f2c9671c62d3f98 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -3625,7 +3625,7 @@ function generatePropTypes(ctx) {
   type ImportantMark = "!" | "!important"
   type WhitespaceImportant = \` \${ImportantMark}\`
   type Important = ImportantMark | WhitespaceImportant
-  type WithImportant<T> = T extends string ? \`\${T}\${Important}\${string}\` : T
+  type WithImportant<T extends U, U = any> = U extends string ? \`\${U}\${Important}\` & { __important?: true } : T;
 
   /**
    * Only relevant when using \`strictTokens\` or \`strictPropertyValues\` in your config.
diff --git a/dist/index.mjs b/dist/index.mjs
index 4eba2eb24aa95ae802a80ca780f38970f81d89a8..2f99787fd92a4a3491773c1d278fa6cd3d1a31a1 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -3589,7 +3589,7 @@ function generatePropTypes(ctx) {
   type ImportantMark = "!" | "!important"
   type WhitespaceImportant = \` \${ImportantMark}\`
   type Important = ImportantMark | WhitespaceImportant
-  type WithImportant<T> = T extends string ? \`\${T}\${Important}\${string}\` : T
+  type WithImportant<T extends U, U = any> = U extends string ? \`\${U}\${Important}\` & { __important?: true } : T;
 
   /**
    * Only relevant when using \`strictTokens\` or \`strictPropertyValues\` in your config.
